### DISCUSSION TITLE
Build for iOS Simulator on arm64

### DISCUSSION THREAD

**Discussion by tsouza** [nodejs-mobile/nodejs-mobile#22] [Ideas]
I've been researching how to build nodejs-mobile for running on iOS Simulator for arm64. My first attempt was modifying the ios build scripts so it would only build nodejs with `--dest-os=ios --dest-cpu=arm64` and then build both frameworks (`iphoneos` and `iphonesimulator`) with `-arch arm64`. But my attempt failed at `ld` command for `-arch arm64 -sdk iphonesimulator` with an error that did not make much sense saying that the nodejs `lib*.a` were built for `iOS` instead of `iOS Simulator`.

So after more research it turns out we can simply transform the binary built for iphone on arm64 into iphonesimulator arm64. The command was taken from [this great article](https://bogo.wtf/arm64-to-sim-dylibs.html):
```
$ xcrun vtool -arch arm64 \
              -set-build-version 7 11.0 11.0 \
              -replace \
              -output NodeMobile.framework/NodeMobile.iphonesimulator \
              NodeMobile.framework/NodeMobile
```
Which produces a new binary that will run in iOS Simulator on arm64. I can confirm this is at least working for me (I am building a swift library on top of nodejs mobile)

I can open a PR with the necessary changes. I think we can use `lipo` to build a version of NodeMobile for both x86_64 and arm64.

**Comment by staltz** [comment]
Thanks for the tip! I'll try this myself, and indeed, I have once used `vtool` to hack the headers of the binary, and it works. I'll look into incorporating this into the nodejs-mobile scripts too, but hopefully in a manner that still supports x64 MacOS hosts.
